src/test/java/com/fasterxml/jackson/integtest/gradle/GradleTest.java

```diff
@@ -0,0 +1,182 @@
+package com.fasterxml.jackson.integtest.gradle;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class GradleTest
+{
+
+    static final String VERSION_QUALIFIER = "rc1"; // currently fails for SNAPSHOT, as not all modules have a snapshot
```

Comment:
This should be `SNAPSHOT` so that the test always runs again the latest snapshot as all other tests in this repository. However, some modules do not have a SNAPSHOT published at the moment. These are:

```
com.fasterxml.jackson.datatype:jackson-datatype-hibernate3
com.fasterxml.jackson.datatype:jackson-datatype-hibernate4
com.fasterxml.jackson.datatype:jackson-datatype-hibernate5
com.fasterxml.jackson.datatype:jackson-datatype-jaxrs
com.fasterxml.jackson.jaxrs:jackson-jaxrs-base
com.fasterxml.jackson.jaxrs:jackson-jaxrs-cbor-provider
com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider
com.fasterxml.jackson.jaxrs:jackson-jaxrs-smile-provider
com.fasterxml.jackson.jaxrs:jackson-jaxrs-xml-provider
com.fasterxml.jackson.jaxrs:jackson-jaxrs-yaml-provider
com.fasterxml.jackson.module:jackson-module-jsonSchema
```

Is this expected? If yes, we can also exclude these modules from the test by removing them from the `JACKSON_MODULES` list below. (But then they will never be tested although they would work with RC and GA versions.)

Reply:
Snapshot builds are automated via Travis, but I think some projects may have fallen through cracks.
I also sometimes manually push snapshots right after release for convenience.

Intent would be for all projects to auto-publish; one concern going forward is that of how to indicate cases where modules are dropped (JSON schema will not be published for 3.x as there are better alternatives; Hibernate module has no maintainers so it may or may not continue).

I'll try to add auto-publishing today; it's some work (just to generate credentials etc). But before that I think I can manually push snapshots for 2.12.0-rc2-SNAPSHOT for these 3 repos.

Reply:
Thanks for pushing the snapshots manually. Unfortunately it does not help as these are `2.12-rc2-SNAPSHOT` and not `2.12-SNAPSHOT`. At least Gradle is not able to discover them when you use a `SNAPSHOT` dependency. (Not sure if this works in Maven, the _SNAPSHOT_ concept has some hard to understand behaviour in its details and there are still a few differences between Maven and Gradle.)

I changed the test now to use `SNAPSHOT`. I don't know how much work it is, but you could publish manually again by temporarily removing the "-rc2" from the versions.

Or we can just wait with merging this until auto-publishing is working, if you plan to work on it anyway, and the version is changed from no longer being 'RC'?

Regarding modules that will be dropped: This test is supposed to only test against the latest version. So if modules are dropped, and you upgrade the integration tests to run against `3.x` (by upgrading the [version in the pom](https://github.com/jjohannes/jackson-integration-tests/blob/e2ed01c6c95fd25891fafa9ec3ecb5e2abedcdb8/pom.xml#L7)) the list here would need to be adjusted manually to reflect the 3.x situation. I suppose that's ok as this is not happening very often. Or do you have an idea how to obtain the list dynamically (maybe we can fetch the BOM and parse it's entries)?

Reply:
Hmmh. So, no, there are no new 2.12.0-SNAPSHOTs (there were old ones, before starting release-candidate process), and I don't think I'll want to go and do this for all projects.
I am still thinking of releasing 2.12.0-rc2 before final 2.12.0 which is why 2.12.0-rc2-SNAPSHOT is released (and there is a whole sorry "2.12.0-rc1 vs 2.12.0.rc1" back and forth problem -- former does not work OSGi users; latter is not recognized as non-release version by some tools like Tidelift -- there apparently is no good way to indicate these... and I've tried both).

I don't think auto-publish would help here, either (although I'll set that up for jax-rs at least); it would produce same versions that I do, I think.

Reply:
I understand. These sound like common problems with snapshots.

I think it makes no sense to run the test against snapshots then. There is no guarantee that a snapshot of all components is available for a given version.

I changed the test to use Gradle to determine the latest release and then test that. RCs also count as release. This means, as soon as a new RC is published, the next time the test runs it will test that one. In the case something broke, the test will still capture it before the final release.

This seems like the best solution for now to me.

Reply:
That does sound reasonable, yes. I think that while it'd be nice in perfect world to check snapshots, use of release candidates should at least reduce risk of broken releases.

And so I think we'll see what 2.12.0-rc2 brings about: this time I will release that (not just rc1), due to more changes going in 2.12 than 2.11 (which was sort of "small" release).

I merged jackson-jr/annotation-support inclusion in jackson-bom; and also made jackson-jaxrs-providers push snapshots via Travis (even if that doesn't matter for this particular issue).